newlib/libc/tinystdio/fgets.c

```diff
@@ -36,15 +36,21 @@ fgets(char *str, int size, FILE *stream)
 {
 	char *cp;
 	int c;
+	int empty_stream_flag = 0; //flag checks if stream is empty or not
 
 	if ((stream->flags & __SRD) == 0 || size <= 0)
 		return NULL;
 
 	size--;
 	for (c = 0, cp = str; c != '\n' && size > 0; size--, cp++) {
-		if ((c = getc(stream)) == EOF)
-			return NULL;
+		if ((c = getc(stream)) == EOF) {
+			if(empty_stream_flag == 0)
```

Comment:
The `gets` code tests the pointer to see if it has moved. Can we do that here by using `if (cp == str)`? That would make the two versions a little more similar and avoid adding another variable to this code.

Reply:
I added the suggested change in the fgets function